robottelo/entities.py

```diff
@@ -2288,25 +2288,60 @@ class Meta(object):
         server_modes = ('sat', 'sam')
 
 
-class SyncPlan(orm.Entity):
+class SyncPlan(orm.Entity, orm.EntityReadMixin, orm.EntityDeleteMixin,
+        orm.EntityCreateMixin):
     """A representation of a Sync Plan entity."""
-    organization = entity_fields.OneToOneField('Organization', required=True)
+    enabled = entity_fields.BooleanField(required=True)
     name = entity_fields.StringField(required=True)
     # how often synchronization should run must be one of: none, hourly, daily,
     # weekly.
     interval = entity_fields.StringField(
-        choices=('none', 'hourly', 'daily', 'weekly'),
+        choices=('hourly', 'daily', 'weekly'),
         required=True,
     )
     # start datetime of synchronization
     sync_date = entity_fields.DateTimeField(required=True)
     description = entity_fields.StringField()
 
-    class Meta(object):
-        """Non-field information about this entity."""
-        api_path = 'katello/api/v2/organizations/:organization_id/sync_plans'
-        server_modes = ('sat')
+    def __init__(self, organization, **kwargs):
+        """Record ``organization_id`` and set ``self.Meta.api_path``."""
+        self.Meta.organization_id = organization
+        self.Meta.api_path = '{0}/sync_plans'.format(
+            Organization(id=organization).path()
+        )
+        super(SyncPlan, self).__init__(**kwargs)
+
+    def read(self, auth=None, entity=None, attrs=None, ignore=()):
+        """Provide a default value for ``entity``.
+
+        By default, :meth:`robottelo.orm.EntityReadMixin.read` provides a
+        default value for ``entity`` like so::
+
+            entity = type(self)()
+
+        However, :class:`SyncPlan` requires that an ``organization_id`` be
+        provided, so this technique will not work. Do this instead::
+
+            entity = type(self)(self.Meta.os_id)
 
+        """
+        # `entity = self` also succeeds. However, the attributes of the object
+        # passed in will be clobbered. Passing in a new object allows this one
+        # to avoid changing state. The default implementation of
+        # `read` follows the same principle.
+        if entity is None:
+            entity = type(self)(self.Meta.organization_id)
+        return super(SyncPlan, self).read(
+            auth, entity, attrs, ignore
+        )
+
+    def create_payload(self):
+        """Creating a ``sync plan`` requires that the attribute be a string
+        value, so we need to convert it to its string representation.
+        """
```

Comment:
All multi-line docstrings that I have seen add an empty line like https://github.com/omaciel/robottelo/blob/api-syncplan/robottelo/entities.py#L179.

The adopted formatting is like below

```python
"""One line"""

"""Multiline docstring ...
continuation ...

"""
```